### ISSUE TITLE
NEDS->velocity() seems to be absolute in translation and relative in rotation 

### ISSUE THREAD

**Issue by radarsat1** [siconos/siconos#163]
I just want to verify that this is correct.  I am currently implementing "joint torques" and have managed to get a bit confused about frames of reference with regards to velocity.  I sort of figured it out but there remains a question that I can simplify here.

Here is simply a cube, rotated 90 degrees, with an initial non-zero angular and linear velocity:
```python
from siconos.mechanics.collision.tools import Contactor
from siconos.io.mechanics_io import Hdf5
import siconos.numerics as Numerics
import numpy as np

with Hdf5() as io:
    io.addPrimitiveShape('Cube', 'Box', (2, 2, 2))
    io.addObject('cube', [Contactor('Cube')], translation=[0, 0, 2],
                 velocity=[10, 0, 0, 1, 0, 0],
                 orientation=[(0,0,1),np.pi/2], # Rotated!
                 mass=1)

with Hdf5(mode='r+') as io:
    io.run(t0=0, T=10, h=0.0005, theta=0.50001,
           Newton_max_iter=20,
           set_external_forces=lambda x: None,
           solver=Numerics.SICONOS_FRICTION_3D_NSGS,
           itermax=100000,
           tolerance=1e-8)
```

If you look at the entries in the `velocity` table in `cube.hdf5`, all values are constant, just as specified:  A linear velocity 10 in the X axis, and an angular velocity 1.0 in the X axis.

However, if you look with `siconos_vview`, you'll see that the cube is *translating* in the X axis, but *rotating* around the Y axis.

**Question**:  It appears that the velocity is represented in *absolute* coordinates for translation, and *relative* (body-frame) coordinates for the angular velocity.  Is this expected?

I suspect it must be related to the `_isMextExpressedInInertialFrame` flag, which I only just discovered.  It is *okay*, I just want to make sure that this is correct and will not change in the future.

It is a bit unexpected from a usage point of view, as it means that for calculating velocity-dependent external forces, for the linear case I have to use velocity() directly, but for the angular case I have to call changeFrameBodyToAbs() on a temporary 3-vector extracted from velocity().

One potential "fix" could be to add unambiguously-named functions angularVelocityRel/Abs() and linearVelocityRel/Abs() for example.

**Comment by radarsat1** [comment]
Just came across the comment in NewtonEulerDS.hpp,

~~~
  /** _twist contains the twist of the Newton Euler dynamical system.
   *  _twist[0:2] : \f$v_G \in \RR^3 \f$ velocity of the center of mass in
   * the inertial frame of reference (world frame).
   *  _twist[3:5] : \f$\Omega\in\RR^3\f$ angular velocity expressed in the body-fixed frame
   */
~~~

So I guess that explains it to some extent, but I still feel this somewhat violates the principle of least surprise.  I leave open the idea of "linear/angularVelocityRel/abs()" functions.. I am currently working on an "add off-center force/torque" type of functionality, and these are the kinds of useful "utility" functions for NewtonEulerDS that would make Siconos easier to use as a physics engine.

**Comment by radarsat1** [comment]
After the 4.1 release I will work on a script to help us identify functions that have too-short documentation or that do not point to the variable that they access.  e.g. docstring for twist() is "get twist", but the real documentation is for the variable _twist.  Even if doxygen helps a little by inserting a pointer to `_twist`, it's too indirect and not visible when reading the hpp files.

**Comment by radarsat1** [comment]
Fixed in 48f4bb0e8a20504fc1851de7855ec4568e6db93c
